### ISSUE TITLE
Next dev not running any code on the client 

### ISSUE THREAD

**Issue by iainbryson** [zeit/next.js#3790]
Hi there,

This is a very fundamental question: should next.js be running code on the client when started in dev mode?

For me, it doesn't. `componentDidMount` and event handler callbacks — none of them run in `dev` mode. They work just fine when doing `next build && next start` and `next export`.

It must be something I'm doing wrong, I assume. But how do I debug this? What's the simplest app which should run React code on the browser with `next dev`?

I've looked at the solutions proposed in #1773 and #2869 and nothing I do with `react-no-ssr` or dynamic imports seems to do anything; the no SSR stuff simply doesn't render anywhere using the dev server.

Thanks!

**Comment by iainbryson** [comment]
No, just used the basic instructions.

**Comment by timneutkens** [comment]
Can you provide a reproduction of this. Sounds like the client side scripts are simply not loaded.

**Comment by iainbryson** [comment]
Thanks for the hint that it _should_ work!  That gave me the confidence to go deeper...

tl;dr: it looks like some weird interaction between a version of next.js (which I copy pasted from a co-worker, shame on me) and the `.jsx` extension.

I can repro by:
1. cloning the `progressive-render` example
1. `cp pages/index.js pages/index2.jsx`
1. changing the `next` version in package.json to `^4.4.0-canary.3`
1. `npm install && npm run dev`

When I do that and reload in Chrome, I get the same error as I do in my project:

![image](https://user-images.githubusercontent.com/1531103/36184201-31b46b54-10e7-11e8-89e9-a0a310d0f606.png)

`net::ERR_ABORTED` is not an obvious symptom, and there was nothing in the `npm run dev` console to indicate a problem.

Anyway, I'm very happy this works as expected!

PS: a clean install of this example project yields this:
```
npm WARN react-no-ssr@1.1.0 requires a peer of react@^0.14.0 || ^15.0.0 but none is installed. You must install peer dependencies yourself.
```

**Comment by timneutkens** [comment]
Try upgrading to `next@latest`. The latest version is v5

**Comment by iainbryson** [comment]
I should have been clearer; without the .jsx extension or the canary build, everything is fine.   What's the best way to pick a stable v5 release?  I'd rather not have `@latest` in the package file...

**Comment by iainbryson** [comment]
Perfect!

Thanks @timneutkens !

**Comment by timneutkens** [comment]
Let me know if that solves it. `jsx` handling is now no different from `js` handling. 👍

**Comment by iainbryson** [comment]
Everything works fine now.
